app/src/main/java/com/keylesspalace/tusky/components/viewthread/ViewThreadViewModel.kt

```diff
@@ -0,0 +1,426 @@
+import com.keylesspalace.tusky.appstore.StatusDeletedEvent
+import com.keylesspalace.tusky.components.timeline.util.ifExpected
+import com.keylesspalace.tusky.db.AccountManager
+import com.keylesspalace.tusky.entity.Filter
+import com.keylesspalace.tusky.entity.Status
+import com.keylesspalace.tusky.network.FilterModel
+import com.keylesspalace.tusky.network.MastodonApi
+import com.keylesspalace.tusky.usecase.TimelineCases
+import com.keylesspalace.tusky.util.toViewData
+import com.keylesspalace.tusky.viewdata.StatusViewData
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.async
+import kotlinx.coroutines.channels.BufferOverflow
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.rx3.asFlow
+import kotlinx.coroutines.rx3.await
+import javax.inject.Inject
+
+class ViewThreadViewModel @Inject constructor(
+    private val api: MastodonApi,
+    private val filterModel: FilterModel,
+    private val timelineCases: TimelineCases,
+    eventHub: EventHub,
+    accountManager: AccountManager
+) : ViewModel() {
+
+    private val _uiState: MutableStateFlow<ThreadUiState> = MutableStateFlow(ThreadUiState.Loading)
+    val uiState: Flow<ThreadUiState>
+        get() = _uiState
+
+    private val _errors = MutableSharedFlow<Throwable>(replay = 0, extraBufferCapacity = 1, onBufferOverflow = BufferOverflow.DROP_OLDEST)
+    val errors: Flow<Throwable>
+        get() = _errors
+
+    var isInitialLoad: Boolean = true
+
+    private val alwaysShowSensitiveMedia: Boolean
+    private val alwaysOpenSpoiler: Boolean
+
+    init {
+        val activeAccount = accountManager.activeAccount
+        alwaysShowSensitiveMedia = activeAccount?.alwaysShowSensitiveMedia ?: false
+        alwaysOpenSpoiler = activeAccount?.alwaysOpenSpoiler ?: false
+
+        viewModelScope.launch {
+            eventHub.events
+                .asFlow()
+                .collect { event ->
+                    when (event) {
+                        is FavoriteEvent -> handleFavEvent(event)
+                        is ReblogEvent -> handleReblogEvent(event)
+                        is BookmarkEvent -> handleBookmarkEvent(event)
+                        is PinEvent -> handlePinEvent(event)
+                        is BlockEvent -> removeAllByAccountId(event.accountId)
+                        is StatusComposedEvent -> handleStatusComposedEvent(event)
+                        is StatusDeletedEvent -> handleStatusDeletedEvent(event)
+                    }
+                }
+        }
+
+        loadFilters()
+    }
+
+    fun loadThread(id: String) {
+        viewModelScope.launch {
+            val contextCall = async { api.statusContext(id) }
+            val statusCall = async { api.statusAsync(id) }
+
+            val contextResult = contextCall.await()
```

Comment:
is this intended that we wait for both before showing anything? It's probably for the best but I wanted to clarify

Reply:
yes, I wanted to avoid the flickering that sometimes occured in the old implementation